ttnn/cpp/ttnn/operations/embedding/device/kernels/dataflow/embedding_ind_tilized.cpp

```diff
@@ -0,0 +1,220 @@
+#if (weight_stick_size_is_power_of_two)
+    constexpr uint32_t log_base_2_of_weights_page_size = get_compile_time_arg_val(7);
+    const InterleavedPow2AddrGen<weights_is_dram> weights = {
+        .bank_base_address = weights_dram_buffer_src_addr,
+        .log_base_2_of_page_size = log_base_2_of_weights_page_size  // TODO(AP): refactor
+    };
+#else
+    const InterleavedAddrGen<weights_is_dram> weights = {
+        .bank_base_address = weights_dram_buffer_src_addr, .page_size = weight_stick_size};
+#endif
+
+    constexpr uint32_t row_length = get_compile_time_arg_val(8);
+    constexpr uint32_t input_block_size_bytes = get_compile_time_arg_val(9);
+
+    constexpr uint32_t cb_id_in0 = 0;
+    constexpr uint32_t cb_id_in1 = 1;
+    constexpr uint32_t cb_id_in2 = 2;
+
+    constexpr uint32_t face_size = 16;
+    constexpr uint32_t tile_height = 32;
+
+#if defined PADDED
+    const std::uint32_t pad_token = get_arg_val<uint32_t>(6);
+    uint64_t pad_noc_addr;
+    {
+        cb_reserve_back(cb_id_in2, 1);
+        uint32_t local_pad_addr = get_write_ptr(cb_id_in2);
+        uint64_t src_noc_addr = get_noc_addr(pad_token, weights);
+        noc_async_read(src_noc_addr, local_pad_addr, weight_stick_size);
+        noc_async_read_barrier();
+        pad_noc_addr = get_noc_addr(local_pad_addr);
+    }
+#elif defined BINARY
+    uint64_t zero_noc_addr, one_noc_addr;
+    {
+        cb_reserve_back(cb_id_in2, 2);
+        uint32_t local_write_addr = get_write_ptr(cb_id_in2);
+        uint64_t src_noc_addr = get_noc_addr(0, weights);
+        noc_async_read(src_noc_addr, local_write_addr, weight_stick_size);
+        zero_noc_addr = get_noc_addr(local_write_addr);
+
+        local_write_addr += weight_stick_size;
+        src_noc_addr = get_noc_addr(1, weights);
+        noc_async_read(src_noc_addr, local_write_addr, weight_stick_size);
+        one_noc_addr = get_noc_addr(local_write_addr);
+
+        noc_async_read_barrier();
+    }
+#endif
+
+    cb_reserve_back(cb_id_in1, 1);
+    uint32_t input_l1_addr = get_write_ptr(cb_id_in1);
+    const uint32_t tile_size_bytes = get_tile_size(cb_id_in1);
+    const DataFormat data_format = get_dataformat(cb_id_in1);
+#if defined BFP16
+    volatile tt_l1_ptr uint16_t* input_l1_ptr = reinterpret_cast<volatile tt_l1_ptr uint16_t*>(input_l1_addr);
+#else
+    volatile tt_l1_ptr uint32_t* input_l1_ptr = reinterpret_cast<volatile tt_l1_ptr uint32_t*>(input_l1_addr);
+#endif
+    // bool printed = false;
+    auto read_block = [&](const uint32_t& token_idx, const uint32_t& width_size, const uint32_t& offset = 0) {
+        cb_reserve_back(cb_id_in0, 1);
+        uint32_t weight_l1_addr = get_write_ptr(cb_id_in0);
+        volatile tt_l1_ptr uint16_t* weight_l1_ptr = reinterpret_cast<volatile tt_l1_ptr uint16_t*>(weight_l1_addr);
+        uint64_t src_noc_addr;
+        uint32_t token = input_l1_ptr[token_idx + offset];
+
+        // I would like to keep this for future debugging purposes
```

Comment:
If we want to keep this around, let's add comments around the magic numbers, like what's 105?